### ISSUE TITLE
Only kv is compressed. Is the size of Q and K inconsistent when attention is calculated?

### ISSUE THREAD

**Issue by CSEEduanyu** [FasterDecoding/SnapKV#10]
Only kv is compressed. Is the size of Q and K inconsistent when attention is calculated?

**Comment by CSEEduanyu** [comment]
@leeyeehoo

**Comment by CSEEduanyu** [comment]
attn_output = self._flash_attention_forward(
        query_states,
        key_states,
        value_states,
        attention_mask,
        q_len,
        dropout=dropout_rate,
        use_sliding_windows=use_sliding_windows,
    )

**Comment by CSEEduanyu** [comment]
attn_output = self._flash_attention_forward(
    query_states,
    key_states,
    value_states,
    attention_mask,
    q_len,
    dropout=dropout_rate,
    use_sliding_windows=use_sliding_windows,
)

**Comment by joyolee** [comment]
> [@leeyeehoo](https://github.com/leeyeehoo) attn_output = self._flash_attention_forward( query_states, key_states, value_states, attention_mask, q_len, dropout=dropout_rate, use_sliding_windows=use_sliding_windows, ) And this is actually a pointer to qkv, how to get the actual k and v to take the length of max_cache_length instead of the length of q_len?


I would ask, is it resolved and how? Thank you
